includes/class-wc-payments-features.php

```diff
@@ -57,4 +57,19 @@ public static function is_sofort_enabled() {
 	public static function is_upe_enabled() {
 		return '1' === get_option( '_wcpay_feature_upe', '0' );
 	}
+
+	/**
+	 * Returns feature flags as an array suitable for display on the front-end.
+	 *
+	 * @return bool[]
+	 */
+	public static function to_array() {
+		return [
+			'groupedSettings' => self::is_grouped_settings_enabled(),
+			'giropay'         => self::is_giropay_enabled(),
```

Comment:
I'm not sure we should be using the giropay/sepa/sofort on the settings React components.
Those feature flags are tied to the related gateway classes (`Giropay_Payment_Gateway`, `Sepa_Payment_Gateway`, `Sofort_Payment_Gateway`), which we tried to decouple from.

The available gateways are returned by the `get_upe_available_payment_methods` (or the `get_enabled_payment_gateways` method on the `UPE_Payment_Gateway` class, which I'll try and consolidate), I think we should be moving forward with that 🤷 

It doesn't look like these feature flags are currently being used by the consumers of `WCPaySettingsContext`, but I just wanted to point this out. Should we remove these attributes from the array?

Reply:
Great point!

> It doesn't look like these feature flags are currently being used by the consumers of WCPaySettingsContext, but I just wanted to point this out. Should we remove these attributes from the array?

I think so. I have added a filter only displaying the flags that are enabled (so if you disable a flag, it will not be present in the `featureFlags` field rather than it being returned as `false`) but since these flags aren't used anywhere anyway, maybe they shouldn't be returned. This leaves with just one flag: `upe`. 😬